### PR TITLE
update website for spark 2.4.2

### PR DESCRIPTION
**Pull request by cloud-fan**
update some links and add news and release note.

### CODE REVIEW

**Comment by cloud-fan** [comment]
cc @srowen @gatorsmile @holdenk @felixcheung @zsxwing

File: releases/_posts/2019-04-23-spark-release-2-4-2.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+tags: []
+status: publish
+type: post
+published: true
+meta:
+  _edit_last: '4'
+  _wpas_done_all: '1'
+---
+
+Spark 2.4.2 is a maintenance release containing stability fixes. This release is based on the branch-2.4 maintenance branch of Spark. We strongly recommend all 2.4.0 users to upgrade to this stable release.
```
Comment (felixcheung) [inline releases/_posts/2019-04-23-spark-release-2-4-2.md]: very small nit - should `all 2.4.0` be `all 2.4.0 and 2.4.1` ?

Comment (cloud-fan) [reply releases/_posts/2019-04-23-spark-release-2-4-2.md]: how about `all 2.4 users`?

File: js/downloads.js [hunk 2]
```diff
@@ -15,14 +15,14 @@ var sources = {pretty: "Source Code", tag: "sources"};
 var hadoopFree = {pretty: "Pre-build with user-provided Apache Hadoop", tag: "without-hadoop"};
 var hadoop2p6 = {pretty: "Pre-built for Apache Hadoop 2.6", tag: "hadoop2.6"};
 var hadoop2p7 = {pretty: "Pre-built for Apache Hadoop 2.7 and later", tag: "hadoop2.7"};
-var scala2p12_hadoopFree = {pretty: "Pre-build with Scala 2.12 and user-provided Apache Hadoop", tag: "without-hadoop-scala-2.12"};
+var scala2p11_hadoopFree = {pretty: "Pre-build with Scala 2.11 and user-provided Apache Hadoop", tag: "without-hadoop-scala-2.11"};
```
Comment (cloud-fan) [inline js/downloads.js:18]: @srowen do we still need it? In the past we provided a scala 2.12 build for users to try out. Now we have moved to scala 2.12, do we need to leave a special scala 2.11 build?

**Comment by zsxwing** [comment]
LGTM

File: downloads.md [hunk 3]
```diff
@@ -26,12 +26,14 @@ $(document).ready(function() {
 
 4. Verify this release using the <span id="sparkDownloadVerify"></span> and [project release KEYS](https://www.apache.org/dist/spark/KEYS).
 
+Note that, Spark 2.3 is pre-built with Scala 2.11, Spark 2.4 is pre-built with Scala 2.12.
```
Comment (zsxwing) [inline downloads.md]: nit: `Spark 2.4` -> `Spark 2.4.2`

**Review by HyukjinKwon — APPROVED** [review]

**Comment by cloud-fan** [comment]
I'm merging it to unblock the release process. Will address comments in the followup if there are any.

**Pull request merged**